src/App.tsx

```diff
@@ -15,36 +17,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+let goods = [...goodsFromServer];
+let isReverseCLicked = false;
+
 export const App: React.FC = () => {
+  const ALPHABETICALLY: string = 'alphabetically';
+  const BY_LENGTH: string = 'by length';
+  const RESET: string = 'reset';
+  const [sortGoods, setSortGoods] = useState('');
+  const [reverse, setReversed] = useState(false);
+
+  function fSort() {
+    if (sortGoods === ALPHABETICALLY) {
+      goods = [...goodsFromServer].sort((goodA, goodB) =>
+        goodA.localeCompare(goodB),
+      );
+    }
+
+    if (sortGoods === BY_LENGTH) {
+      goods = [...goodsFromServer].sort(
+        (goodA, goodB) => goodA.length - goodB.length,
+      );
+    }
+
+    if (reverse) {
+      goods.reverse();
+    }
+  }
```

Comment:
[CODE KNOWLEDGE] You're directly manipulating the `goods` array. This is not recommended in React. Instead, use the `setSortGoods` function provided by the `useState` hook to update your state. This will ensure that your component re-renders correctly when the state changes.